### ISSUE TITLE
readFile not reading entire DTX

### ISSUE THREAD

**Issue by tpvidigal** [akiomik/dtx2midi#2]
Hi @akiomik !
Sorry reviving a so old project of yours, but it's exactly what I wanted :)
However, I can't manage to run it properly.
I'm trying to generate the MIDI of the following file from the [DTX Starter Pack](http://chanmori.net/dtxir/ranking.php?disp=box&hash=55e6e390db7dbe466ee5ee516f1e57c8&section=Drums).

[colorful_adv.zip](https://github.com/akiomik/dtx2midi/files/4874961/colorful_adv.zip)

I'm not an expert in Haskell, but I think "readFile" is not reading the entire file for some reason...
I made a adhoc test printing what was being read from the file:
`toMIDI lines = do`
`    print (lines)`

And the only thing that was printed was:
`[LineComment "",LineComment "",LineHeader (Header {headerKey = "TITLE", headerChannel = "", headerValue = "COLORFUL"}),LineComment "",LineHeader (Header {headerKey = "ARTIST", headerChannel = "", headerValue = "SWEEZ & Meine Meinung"}),LineComment ""]`

It seems that it only got the header info...
Any clues?

**Comment by akiomik** [comment]
> Sorry reviving a so old project of yours, but it's exactly what I wanted :)

No problem, sitting issues and PRs are welcome 😉 
(But I don't know if I can support it because I haven't written Haskell for a long time tho...)

> It seems that it only got the header info...

This caused by the parser detects unkown header key, so I added missing header keys in your dtx file (7d0d5b04c9170247f5528d5ce391708f3e5691fa).
Since it seems to be working properly, could you try to generate it again?

**Comment by tpvidigal** [comment]
Definitely the sheet has too many pauses, but it worked!
Thank you, Akiomik. This will really help me to study the musics :)
Bug fixed!

![image](https://user-images.githubusercontent.com/11837773/86549350-c47d8400-bf15-11ea-8b9e-7442ec8e04a7.png)

My only suggestion is to improve the program to simply ignore unknown header key (if possible).
I tested on another file and it failed again, but I removed some unnecessary header keys and it worked.
